content/docs/tina-cloud/client.md

```diff
@@ -0,0 +1,229 @@
+      getSiteNavsDocument(relativePath: "site-nav.md") {
+        data {
+          items {
+            title
+            link
+          }
+        }
+      }
+      # this will generate a separate Tina Form
+      getPostsDocument(relativePath: $relativePath) {
+        data {
+          title
+        }
+      }
+    }
+  }
+`
+```
+
+### Form customization
+
+If you'd like to control the output of those forms, tap into the `formify` callback:
+
+```tsx
+import { useGraphqlForms } from 'tina-graphql-gateway'
+import { Form, GlobalFormPlugin, useCMS } from 'tinacms'
+
+const [payload, isLoading] = useGraphqlForms({
+  query,
+  formify: ({ formConfig, createForm, skip }) => {
+    if (formConfig.id === 'getSiteNavsDocument') {
+      const form = new Form(formConfig)
+      // The site nav will be a global plugin
+      cms.plugins.add(new GlobalFormPlugin(form))
+      return form
+    }
+
+    return createForm(formConfig)
+  },
+  variables: { relativePath: `${props.filename}.md` },
+})
+
+// or to skip the nav from creating a form altogether:
+const [payload, isLoading] = useGraphqlForms({
+  query,
+  formify: ({ formConfig, createForm, skip }) => {
+    if (formConfig.id === 'getSiteNavsDocument') {
+      return skip()
+    }
+
+    return createForm(formConfig)
+  },
+  variables: { relativePath: `${props.filename}.md` },
+})
+```
+
+### Field customization
+
+Since your forms are built automatically, `formify` can also be used to customize fields:
+
+```tsx
+const [payload, isLoading] = useGraphqlForms({
+  query,
+  formify: ({ formConfig, createForm, skip }) => {
+    return createForm({
+      ...formConfig,
+      fields: formConfig.fields.map(field => {
+        if (field.name === 'title') {
+          // replace `text` with `textarea`
+          field.component = 'textarea'
+        }
+        return field
+      }),
+    })
+  },
+  variables: { relativePath: `${props.filename}.md` },
+})
+```
+
+## Authentication with Tina Cloud
+
+While this package comes with low-level APIs for authentication with Tina Cloud, the easiest way to get started is to use the `TinaCloudAuthWall` component, which will prevent children from rendering until a valid session has been established with Tina Cloud.
+
+```tsx
+import { TinaCloudAuthWall, Client } from "tina-graphql-gateway";
+
+const TinaWrapper = ({ children }) => {
+  const cms = React.useMemo(() => {
+    return new TinaCMS({
+      apis: {
+        tina: new Client({
+          // config
+        })
+      },
+      ...
+    });
+  }, []);
+
+  return <TinaCloudAuthWall cms={cms}>{children}</TinaCloudAuthWall>;
+};
+```
+
+> Note: when using the LocalClient, TinaCloudAuthWall won't display a login screen, there is no authentication for the local GraphQL server.
+
+### Authenticating without `TinaCloudAuthWall`
+
+You can also authenticate with the `Client` directly:
+
+```ts
+const client = new Client({
+  // config
+})
+
+const EditSiteButton = () => {
```

Comment:
I think this code will only work if you wrap your app with Tina. (isn't this a pattern we are trying to drift from?)

Saying this I cant think of a better code sample unless we add the `EditStateProvider` to core